examples/tf-native/ec/hcl/deployment.tf

```diff
@@ -0,0 +1,35 @@
+data "ec_stack" "latest" {
```

Comment:
Should we add the `required_providers` here?
```
terraform {
  required_providers {
    ec = {
      source = "elastic/ec"
      version = "0.2.1"
    }
  }
}
```

I have run Configuration, it failed.

```
➜  chart git:(add_ec_support) ✗ k logs -f ec-deployment-apply-6v2hx

Initializing the backend...

Initializing provider plugins...
- Finding latest version of hashicorp/ec...
╷
│ Error: Failed to query available provider packages
│
│ Could not retrieve the list of available versions for provider
│ hashicorp/ec: provider registry registry.terraform.io does not have a
│ provider named registry.terraform.io/hashicorp/ec
│
│ All modules should specify their required_providers so that external
│ consumers will get the correct providers when using a module. To see which
│ modules are currently depending on hashicorp/ec, run the following command:
│     terraform providers
╵
```

After added the part, `terraform` can download the ec binary.

```
Initializing the backend...

Initializing provider plugins...
- Finding elastic/ec versions matching "0.2.1"...
- Installing elastic/ec v0.2.1...
- Installed elastic/ec v0.2.1 (signed by a HashiCorp partner, key ID 7FE579EDEC6DAA7B)

Partner and community providers are signed by their developers.
If you'd like to know more about provider signing, you can read about it here:
https://www.terraform.io/docs/cli/plugins/signing.html

Terraform has created a lock file .terraform.lock.hcl to record the provider
selections it made above. Include this file in your version control repository
so that Terraform can guarantee to make the same selections by default when
you run "terraform init" in the future.

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
```

Reply:
Yes. I have updated the example to include required_providers.

Reply:
I used the trial account of EC Cloud and I got the following issue.

```
➜  hcl git:(add_ec_support) t apply -auto-approve
╷
│ Error: failed retrieving the specified stack version: 1 error occurred:
│ 	* api error: root.unauthenticated: The supplied authentication is invalid
│
│
│
│   with data.ec_stack.latest,
│   on deployment.tf line 10, in data "ec_stack" "latest":
│   10: data "ec_stack" "latest" {
```

Reply:
I regenerated the API key per you updated link, and it worked.

```
Plan: 1 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + ES_HTTPS_ENDPOINT = (known after apply)
  + ES_PASSWORD       = (sensitive value)
ec_deployment.project: Creating...
╷
│ Error: failed creating deployment: 2 errors occurred:
│ 	* api error: deployments.invalid_autoscaling_config: Invalid autoscaling configuration in deployment: [Autoscaling only supported for accounts with paid subscription]
│ 	* set "request_id" to "bqq46r4f2ys3hnk74bqnsskphadpohf7r91v0lysjcmgba4ppumlj9t01couwzzz" to recreate the deployment resources
│
│
│
│   with ec_deployment.project,
│   on deployment.tf line 15, in resource "ec_deployment" "project":
│   15: resource "ec_deployment" "project" {
│
```